### ISSUE TITLE
ValueError: <MatchTree: root= ... is not in list (guessit 1.0.2)

### ISSUE THREAD

**Issue by medariox** [guessit-io/guessit#260]
Over at SR we have a few users reporting this error. We use subliminal develop branch with guessit 1.0.2. Not sure what the cause of this error is, but seems like guessit needs to handle that exception.

Two issue reports:
```
POSTPROCESSER :: Traceback (most recent call last):
  File "C:\SickRage\SickRage\sickbeard\subtitles.py", line 199, in download_subtitles
    matches = subtitle.get_matches(video, hearing_impaired=False)
  File "C:\SickRage\SickRage\lib\subliminal\providers\opensubtitles.py", line 69, in get_matches
    matches |= guess_matches(video, guess_episode_info(self.movie_release_name + '.mkv'))
  File "C:\SickRage\SickRage\lib\guessit\__init__.py", line 370, in guess_episode_info
    return guess_file_info(filename, info=info, options=options, type='episode', **kwargs)
  File "C:\SickRage\SickRage\lib\guessit\__init__.py", line 302, in guess_file_info
    result.append(_guess_filename(filename, options, **kwargs))
  File "C:\SickRage\SickRage\lib\guessit\__init__.py", line 114, in _guess_filename
    mtree = _build_filename_mtree(filename, options=options, **kwargs)
  File "C:\SickRage\SickRage\lib\guessit\__init__.py", line 127, in _build_filename_mtree
    mtree = IterativeMatcher(filename, options=merged_options, **kwargs)
  File "C:\SickRage\SickRage\lib\guessit\matcher.py", line 120, in __init__
    self._process(transformer, False)
  File "C:\SickRage\SickRage\lib\guessit\matcher.py", line 138, in _process
    transformer.process(self.match_tree, self.options)
  File "C:\SickRage\SickRage\lib\guessit\transfo\guess_release_group.py", line 223, in process
    GuessFinder(self.guess_release_group, None, self.log, options).process_nodes(mtree.unidentified_leaves())
  File "C:\SickRage\SickRage\lib\guessit\matcher.py", line 227, in process_nodes
    self.process_node(node)
  File "C:\SickRage\SickRage\lib\guessit\matcher.py", line 286, in process_node
    self.process_node(node, partial_span=relative_span, skip_nodes=next_skip_nodes)
  File "C:\SickRage\SickRage\lib\guessit\matcher.py", line 241, in process_node
    matcher_result = self.guess_func(string, node, self.options)
  File "C:\SickRage\SickRage\lib\guessit\transfo\guess_release_group.py", line 175, in guess_release_group
    if self.validate_node(leaf, node, True):
  File "C:\SickRage\SickRage\lib\guessit\transfo\guess_release_group.py", line 143, in validate_node
    if not self.validate_next_leaves(node):
  File "C:\SickRage\SickRage\lib\guessit\transfo\guess_release_group.py", line 126, in validate_next_leaves
    next_leaf = node.root.next_leaf(node)
  File "C:\SickRage\SickRage\lib\guessit\matchtree.py", line 318, in next_leaf
    return self._other_leaf(leaf, +1)
  File "C:\SickRage\SickRage\lib\guessit\matchtree.py", line 322, in _other_leaf
    index = leaves.index(leaf) + offset
ValueError: <MatchTree: root=-FQM.Live.Free.or.Twi-hard> is not in list
 [ed25b12]
```

```
FINDSUBTITLES :: Traceback (most recent call last):
  File "H:\usr\local\sickrage\sickbeard\subtitles.py", line 177, in download_subtitles
    matches = subtitle.get_matches(video, hearing_impaired=sickbeard.SUBTITLES_HEARING_IMPAIRED)
  File "H:\usr\local\sickrage\lib\subliminal\providers\opensubtitles.py", line 69, in get_matches
    matches |= guess_matches(video, guess_episode_info(self.movie_release_name + '.mkv'))
  File "H:\usr\local\sickrage\lib\guessit\__init__.py", line 370, in guess_episode_info
    return guess_file_info(filename, info=info, options=options, type='episode', **kwargs)
  File "H:\usr\local\sickrage\lib\guessit\__init__.py", line 302, in guess_file_info
    result.append(_guess_filename(filename, options, **kwargs))
  File "H:\usr\local\sickrage\lib\guessit\__init__.py", line 114, in _guess_filename
    mtree = _build_filename_mtree(filename, options=options, **kwargs)
  File "H:\usr\local\sickrage\lib\guessit\__init__.py", line 127, in _build_filename_mtree
    mtree = IterativeMatcher(filename, options=merged_options, **kwargs)
  File "H:\usr\local\sickrage\lib\guessit\matcher.py", line 120, in __init__
    self._process(transformer, False)
  File "H:\usr\local\sickrage\lib\guessit\matcher.py", line 138, in _process
    transformer.process(self.match_tree, self.options)
  File "H:\usr\local\sickrage\lib\guessit\transfo\guess_release_group.py", line 223, in process
    GuessFinder(self.guess_release_group, None, self.log, options).process_nodes(mtree.unidentified_leaves())
  File "H:\usr\local\sickrage\lib\guessit\matcher.py", line 227, in process_nodes
    self.process_node(node)
  File "H:\usr\local\sickrage\lib\guessit\matcher.py", line 286, in process_node
    self.process_node(node, partial_span=relative_span, skip_nodes=next_skip_nodes)
  File "H:\usr\local\sickrage\lib\guessit\matcher.py", line 241, in process_node
    matcher_result = self.guess_func(string, node, self.options)
  File "H:\usr\local\sickrage\lib\guessit\transfo\guess_release_group.py", line 175, in guess_release_group
    if self.validate_node(leaf, node, True):
  File "H:\usr\local\sickrage\lib\guessit\transfo\guess_release_group.py", line 143, in validate_node
    if not self.validate_next_leaves(node):
  File "H:\usr\local\sickrage\lib\guessit\transfo\guess_release_group.py", line 126, in validate_next_leaves
    next_leaf = node.root.next_leaf(node)
  File "H:\usr\local\sickrage\lib\guessit\matchtree.py", line 318, in next_leaf
    return self._other_leaf(leaf, +1)
  File "H:\usr\local\sickrage\lib\guessit\matchtree.py", line 322, in _other_leaf
    index = leaves.index(leaf) + offset
ValueError: <MatchTree: root=.ZiLLA.There's.a.Whole.Slew.of.Ladies.with.Bad.Things.to.Say.About.th> is not in list
 [be60487]
``` 

Thank you.

**Comment by Toilal** [comment]
FYI, subliminal has just merged my PR to use guessit 2. So the issue is probably fixed now if you upgrade subliminal to latest develop version.

**Comment by labrys** [comment]
He's already testing it :)

**Comment by Toilal** [comment]
Great, please give feedbacks, hope it will work nicely :)

**Comment by Toilal** [comment]
It let this issue open and will try to fix it for 1.x users.

**Comment by Toilal** [comment]
Do you have value of `self.movie_release_name` variable ?

**Comment by Toilal** [comment]
I can reproduce the issue with this entry : `The.Clone.Wars.S06E04.720p.WEBRip.FR.W4F.FR.NoTAG.mkv`

**Comment by Toilal** [comment]
Released in `1.0.3`
